### DISCUSSION TITLE
GitHub Copilot authentication issues

### DISCUSSION THREAD

**Discussion by mp-datafleet** [community/community#58876] [Copilot Conversations]
### Select Topic Area

Bug

### Body

Hi guys, 

I've been given a license for Github Copilot by my company and after installing the Github Copilot extension in VS Code, I've been unable to connect to my Github account.

In fact, after activating the extension, I'm not given the chance to log in to VS Code. I just get the following error message:

`GitHub Copilot could not connect to server. Extension activation failed: "Timed out waiting for authentication provider to register"`

<img width="577" alt="Screen Shot 2023-06-23 at 9 57 01 AM" src="https://github.com/community/community/assets/106626994/0c4cd1e9-9c2c-4d98-9bdb-d1cbb0a23c69">

I haven't seen any other examples of this issue thus far, so I'm hoping someone could provide me with some guidance.

Thanks in advance!

**Comment by Akash1134** [comment]
Hi @mp-datafleet ,

Sorry that you're having trouble using Copilot. 

- To start with, can you confirm whether you have completed the steps to install and setup the Copilot extension in VS Code according to our documentation? - [Installing the GitHub Copilot extension for Visual Studio Code](https://docs.github.com/en/copilot/quickstart#installing-the-github-copilot-extension-for-visual-studio-code)
- The error message GitHub Copilot could not connect to server. Extension activation failed: "Timed out waiting for authentication provider to register" is often caused by the built-in GitHub Authentication extension becoming disabled.
  - You can find this extension by navigating to the extensions tab in VS Code and entering the search query **@builtin** GitHub Authentication - Once you have found the GitHub Authentication extension, ensure that it is updated and enabled. 
  - Please then restart VS Code and go through the steps authorise Visual Studio Code in your GitHub account and sign in to GitHub in Visual Studio Code again when prompted.
- On a related note, can you also confirm whether you are you using Copilot behind a firewall, VPN, or other technology that may be intercepting your network connections? As this can, on occasion, interrupt authentication of GitHub Copilot.

Thanks!

**Comment by mp-datafleet** [comment]
First, I followed the directions to install and set up Github Copilot's VS Code extension up until this point:
`If you have not previously authorized Visual Studio Code in your GitHub account, you will be prompted to sign in to GitHub in Visual Studio Code.

If you have previously authorized Visual Studio Code in your GitHub account, GitHub Copilot will be automatically authorized.`

VS Code did not prompt me to sign in to Github upon installing and activating the extension. I've disabled, uninstalled, reinstalled, and reactivated the extension numerous times, and still no prompt to sign in.

Second, the VS Code extension is currently installed AND activated.

Third, as far as I know, I am not using any VPN or firewall. I'm on a personal device on my personal internet service.

**Comment by skitchpatterson3** [comment]
In case it helps others:

TL;DR --> Make sure you have set a value for the "**Suggestions matching public code (duplication detection filter)**" setting in the Privacy section of the [Copilot settings page](https://github.com/settings/copilot).

I fought with authentication issues in both Visual Studio 2022 and in VS Code for **_days_** before emailing GitHub support. In the GitHub Copilot drop menu in my Visual Studio IDE, the "Open Chat Window" option was disabled for me. I tried signing out and signing back in with my GitHub account countless times, and each time when the browser window opened to authorize Visual Studio to access my account, I would always see the "Success!" message, yet the Copilot Chat window or panel kept displaying buttons to "Sign up for Copilot Free" or "Add a different GitHub account" or (eventually) to "Refresh credentials".

In VS Code, the Output window logged several messages that didn't point right to the issue, but were symptoms of it:

- [auth] Invalid copilot token: missing token: 403
- [error] GitHub Copilot could not connect to server. Extension activation failed: “Failed to get copilot token”
- [warning] [LanguageModelAccess] LanguageModel/Embeddings are not available without auth token
- [error] Error: Failed to get copilot token
- [error] CopilotAuthError:
at k$.fetchCopilotTokenEnvelope (c:\Users\\(_removed_)\\.vscode\extensions\github.copilot-1.292.0\lib\src\auth\copilotTokenManager.ts:87:19)

GitHub support replied back within a matter of hours (thank you!), informing me that one or more of my Copilot settings was not configured properly. I had assumed everything was okay because I was able to use Copilot in a web browser without any issues at all.

However, on my [Copilot settings page](https://github.com/settings/copilot), there was a yellow banner at the top with the message, "Your GitHub Copilot setup is incomplete. Select your preferences below to start using GitHub Copilot." It seemed an odd message to me, since the page showed no other field validation messages.

![image](https://github.com/user-attachments/assets/707b42ce-913e-4950-ac87-988d9b4f7b1d)

The issue (for me) turned out to be an unset value for the "**Suggestions matching public code (duplication detection filter)**" setting in the Privacy section of the Copilot settings page.

![image](https://github.com/user-attachments/assets/a72c3f19-e87e-41a2-b7cf-6f558a766994)

I chose a value and restarted VS Code and Visual Studio 2022. To my great relief, both applications now successfully connect with Copilot and the Copilot Chat Window works now.
